src/soroban.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Soroban helper class
+ * formatting, parsing, and etc
+ * @class Soroban
+ */
+export class Soroban {
+  /**
+   * All arithmetic inside the contract is performed on integers to
+   * avoid potential precision and consistency issues of floating-point
+   *
+   * This function takes the smart contract value and its decimals (if the token has any) and returns a display value
+   * @param {bigint | number | string} amount - the token amount you want to display
+   * @param {number} decimals - specify how many decimal places a token has
+   * @returns {string} - display value
+   */
+  static formatTokenAmount(amount, decimals) {
+    let formatted = amount.toString();
+
+    if (decimals > 0) {
+      formatted = (amount / 10 ** decimals).toFixed(decimals).toString();
```

Comment:
thoughts? @Shaptic

Reply:
This is an interesting approach, but I'm worried it won't preserve enough precision in the `number` case?

Separate question: is `amount` expected to be a whole number? 

Also, `.toFixed()` returns a string already.

Reply:
`amount` is expected to be a whole number since it's the value that's used on smart contract. Removed `.toString();`

Reply:
should we validate that, then? e.g. `BigInt(amount)` will throw if amount is a decimal or checking `formatted.indexOf('.') === -1`